### ISSUE TITLE
Enable torch.nn.scaled_dot_product_attention

### ISSUE THREAD

**Issue by inatatsu** [torch-spyre/torch-spyre#484]
### What feature would you like to be added?

Run [`torch.nn.scaled_dot_product_attention`](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html) on Spyre.

### Why is this feature needed?

This operator is used by important models.

**Integrations**
- [`attn_output = torch.nn.functional.scaled_dot_product_attention(query, key, value, attn_mask=attention_mask, dropout_p=dropout, scale=scaling, is_causal=is_causal, **sdpa_kwargs,)`](https://github.com/huggingface/transformers/blob/v4.57.1/src/transformers/integrations/sdpa_attention.py#L96-L105)

**Granite Speech**
- [`out = F.scaled_dot_product_attention(query_states, key_states, value_states, attn_mask=pos_attn, scale=self.scale)`](https://github.com/huggingface/transformers/blob/v4.57.1/src/transformers/models/granite_speech/modeling_granite_speech.py#L176-L178)
### Any alternative solution that you may want to share?

None.

**Comment by inatatsu** [comment]
Closing since #424 already tracks this issue. Further investigations will be posted on #424.
